SU2_CFD/src/solver_direct_mean.cpp

```diff
@@ -16664,17 +16666,20 @@ void CNSSolver::BC_ConjugateHeat_Interface(CGeometry *geometry, CSolver **solver
       /*--- Compute the normal gradient in temperature using Twall ---*/
 
       There = nodes->GetTemperature(Point_Normal);
-      Tconjugate = GetConjugateHeatVariable(val_marker, iVertex, 0);
-
-      HF_FactorHere = thermal_conductivity*config->GetViscosity_Ref()/dist_ij;
-      HF_FactorConjugate = GetConjugateHeatVariable(val_marker, iVertex, 2);
+      Tconjugate = GetConjugateHeatVariable(val_marker, iVertex, 0)/Temperature_Ref;
 
-      Twall = (There*HF_FactorHere + Tconjugate*HF_FactorConjugate)/(HF_FactorHere + HF_FactorConjugate);
+      if ((config->GetKind_CHT_Coupling() == AVERAGED_TEMPERATURE_NEUMANN_HEATFLUX)
+          || (config->GetKind_CHT_Coupling() == AVERAGED_TEMPERATURE_ROBIN_HEATFLUX)) {
 
-      // this will be changed soon...
-      Twall = GetConjugateHeatVariable(val_marker, iVertex, 0);
+        HF_FactorHere = thermal_conductivity*config->GetViscosity_Ref()/dist_ij;
+        HF_FactorConjugate = GetConjugateHeatVariable(val_marker, iVertex, 2);
+        Twall = (There*HF_FactorHere + Tconjugate*HF_FactorConjugate)/(HF_FactorHere + HF_FactorConjugate);
+      }
+      else {
```

Comment:
To give a reason why: I tend to `grep` through the repository quite often for keywords to get an overview and if an option is implicitly taken in an `else` one cannot find it directly. (One can always grep the `Get`ter function of course)

Reply:
Hmm alright, I'll add the error message, it's a bit more safe..

Reply:
@pcarruscag is a `switch` statement better, or what is the best way to resolve that? In this case it is only if..else if..else.

Reply:
Mind you that most times this is not a super important optimization, I do show a case in 789 where it is.
But anyway, ideally error conditions will have been handled upstream and one does not need to include error handling code in hot regions (i.e. loops) where it would pollute the instruction cache. In which case if the purpose is to document just add what the "else" implies as a comment. Switches are for "out of many options choose one kind of deal".
The reason why simple if/else code might work better is not (just) the reduction of logical checks, as within a loop the cpu will start predicting the right branch to take. The real advantage is if instead of branching the compiler manages to use conditional move instructions instead.

Reply:
There goes another day me figuring and trying out what conditional move instructions are 🙈 
I actually wanted to argue with branch prediction that it shouldn't matter that much.... but now I can learn sth new 📚